presto-main/src/main/java/com/facebook/presto/operator/ScanFilterAndProjectOperator.java

```diff
@@ -235,8 +235,9 @@ public Page getOutput()
 
     private Page processColumnSource()
     {
-        if (!finishing) {
-            int rowsProcessed = cursorProcessor.process(operatorContext.getSession().toConnectorSession(), cursor, ROWS_PER_PAGE, pageBuilder);
+        DriverYieldSignal yieldSignal = operatorContext.getDriverContext().getYieldSignal();
+        if (!finishing && !yieldSignal.shouldYield()) {
+            int rowsProcessed = cursorProcessor.process(operatorContext.getSession().toConnectorSession(), yieldSignal, cursor, ROWS_PER_PAGE, pageBuilder);
```

Comment:
Do we need `ROWS_PER_PAGE` anymore?  IIRC it was added to prevent run away processing when you have a highly selective filter.  Now that we have the yield signal, do we need another arbitrary stopping condition?